lib/templates/keyfns.ts

```diff
@@ -74,14 +74,16 @@ export function inputParamKeyFn(slot : InputParamSlot) {
 
 export function paramKeyFn(slot : ParamSlot) {
     const id = slot.schema.getArgument('id');
+    const symmetric = slot.schema.getArgument(slot.name)?.getImplementationAnnotation('symmetric');
     return {
         functionName: slot.schema.qualifiedName,
         type: slot.type,
         is_numeric: slot.type.isNumeric(),
         elem: slot.type instanceof Type.Array ? slot.type.elem as Type : null,
         is_numeric_elem: slot.type instanceof Type.Array ? (slot.type.elem as Type).isNumeric() : false,
         associatedIdType: id && !id.is_input ? id.type : null,
-        filterable: slot.filterable
+        filterable: slot.filterable,
+        symmetric: symmetric ? (symmetric as Ast.BooleanValue).value : false 
```

Comment:
This doesn't look right. getImplementationAnnotation calls toJS on the annotation, so symmetric will be a boolean or undefined.